doc/new_boilerplate_migration.md

```diff
@@ -0,0 +1,635 @@
+    BarPropsMixin: $BarPropsMixin.meta,
+  });
+}
+```
+
+Over_react updates:
+
+```dart
+// A new field in component base class
+class UiComponent2 ... {
+  /// A collection of metadata for the prop fields in all prop mixins
+  /// used by the props class of this component.
+  PropsMetaCollection get propsMeta => null;
+}
+```
+
+A new class in over_react:
+
+```dart
+/// A collection of metadata for the prop fields in all prop mixins
+/// used by a given component.
+///
+/// See [PropsMeta] for more info.
+class PropsMetaCollection implements PropsMeta {
+  final Map<Type, PropsMeta> _metaByMixin;
+
+  const PropsMetaCollection(this._metaByMixin);
+
+  /// Returns the metadata for only the prop fields declared in [mixinType].
+  PropsMeta forMixin(Type mixinType) {
+    final meta = _metaByMixin[mixinType];
+    assert(meta != null, 
+        'No meta found for $mixinType;'
+        'it likely isn\'t mixed in by the props class.')
+    return meta ?? const PropsMeta(fields: [], keys: []);
+  }
+
+  // PropsMeta overrides
+
+  @override
+  List<String> get keys => 
+        _metaByMixin.values.expand((meta) => meta.keys).toList();
+
+  @override
+  List<PropDescriptor> get fields => 
+        _metaByMixin.values.expand((meta) => meta.fields).toList();
+
+  @override
+  List<PropDescriptor> get props => fields;
+}
+```
+
+##### Props MapViews
+
+Since props mixins can only be consumed by other generated code, the 
+existing props map view consumption pattern, whereby props mixins are 
+consumed in user-authored MapView subclasses, cannot be supported.
+
+Instead, props map views will be declared similarly to a component, 
+with a factory and props mixin/class, but no component. 
+
+```diff
+import 'package:over_react/over_react.dart';
+
++ part 'foo.over_react.g.dart';
+
+- class FooPropsMapView extends UiPropsMapView with SomeOtherPropsMixin {
+-   FooPropsMapView(Map backingMap) : super(backingMap);
+- }
+
++ UiFactory<FooMapViewProps> FooMapView = $FooMapView; // ignore: undefined_identifier
+
++ class FooMapViewProps = UiProps with SomeOtherPropsMixin;
+
+usage() {
+-   final mapView = FooPropsMapView(someExistingMap);
++   final mapView = FooMapView(someExistingMap);
+}
+```
+
+## Function Components
+
+### Function Component Constraints 
+
+Includes all of [the constraints listed in the Boilerplate Updates section](#constraints), 
+ignoring parts about backwards-compatibility.
+
+* Should be as visually uncluttered as possible.
+
+* Should not wrap excessively for longer component names.
+
+* Should be easy to transition between having and not having default 
+props, and boilerplate shouldn't change shape drastically when doing so.
+
+* Function calls using generated functions should be avoided since they 
+don't allow generic type inference of the `props` arg in the function 
+closure.
+
+### Design
+
+```dart
+import 'package:over_react/over_react.dart';
+
+part 'foo.over_react.g.dart';
+
+UiFactory<FooProps> Foo = uiFunctionComponent(
```

Comment:
I thought we decided against LHS typing?

```suggestion
final Foo = uiFunctionComponent<FooProps>(
```

Reply:
I got this from the current RFD. Greg said we need the LHS typing because otherwise it won't be statically typed in some repos.

Reply:
@greglittlefield-wf is this the case?

Reply:
Yeah, the idea is that LHS typing
1. keeps things consistent with non-function components, which need the LHS typing to statically analyze since the RHS is `dynamic` before it's generated
2. is compatible with the type_annotate_public_apis Effective Dart lint

across repos that have